src/shmem_team.c

```diff
@@ -307,12 +308,15 @@ int shmem_internal_team_split_strided(shmem_internal_team_t *parent_team, int PE
 
     shmem_internal_team_release_psyncs(parent_team, SYNC);
 
-    if (my_pe >= 0 && myteam != NULL && myteam->psync_idx == -1)
-        RAISE_ERROR_MSG("Team split strided failed: child <%d, %d, %d>, parent <%d, %d, %d>\n",
-                         global_PE_start, PE_stride, PE_size,
-                         parent_team->start, parent_team->stride, parent_team->size);
-    else
+    /* If no team was available, print some team triplet info and return -1. */
+    if (my_pe >= 0 && myteam != NULL && myteam->psync_idx == -1) {
+        RAISE_WARN_MSG("Team split strided failed: child <%d, %d, %d>, parent <%d, %d, %d>\n",
```

Comment:
Should this instead be a `DEBUG_MSG`?

Reply:
Yes, that would be better. Also, why are we returning -1? Anything that is nonzero is failure, so -1 is not worse than 1. Perhaps we should (at least internal to SOS) introduce actual error codes and return those? Please use a separate PR if you want to introduce error codes (take a look at Linux `err.h` if you want inspiration).

Reply:
No reason in particular for -1, it's just a placeholder for future error codes.  Perhaps we should return distinct nonzero values for the different failure cases?

Yeah, that sounds good - I will look into error codes.  Are you suggestion we _use_ `errno` (perhaps `EAGAIN` or similar in this case), or create our own "sos_errno" _similar_ to `err.h` (and/or `fi_errno`)?

Reply:
My suggestion is to make our own header with SOS error codes. The header won't be exposed to the user yet, but it will be helpful to have meaningful values when debugging. Errno is problematic with threading.